pkg/operator/certrotationcontroller/certrotationcontroller.go

```diff
@@ -118,7 +118,86 @@ func newCertRotationController(
 		operatorClient,
 		eventRecorder,
 	)
+	ret.certRotators = append(ret.certRotators, certRotator)
 
+	certRotator = certrotation.NewCertRotationController(
+		"KubeControllerManagerLocalhostServing",
+		certrotation.RotatedSigningCASecret{
+			Namespace:              operatorclient.OperatorNamespace,
+			Name:                   "kcm-localhost-serving-signer",
+			Validity:               10 * 365 * rotationDay,
+			Refresh:                8 * 365 * rotationDay, // this means we effectively do not rotate
+			RefreshOnlyWhenExpired: refreshOnlyWhenExpired,
+			Informer:               kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().Secrets(),
+			Lister:                 kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().Secrets().Lister(),
+			Client:                 secretsGetter,
+			EventRecorder:          eventRecorder,
+		},
+		certrotation.CABundleConfigMap{
+			Namespace:     operatorclient.TargetNamespace,
+			Name:          "kcm-localhost-serving-ca",
+			Informer:      kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps(),
+			Lister:        kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Lister(),
+			Client:        configMapsGetter,
+			EventRecorder: eventRecorder,
+		},
+		certrotation.RotatedSelfSignedCertKeySecret{
+			Namespace:              operatorclient.TargetNamespace,
+			Name:                   "kcm-localhost-serving-cert",
+			Validity:               850 * rotationDay, // 28 months - kcm-cpc-reverse-proxy-serving-cert has 26 months
+			Refresh:                425 * rotationDay, // 14 months - kcm-cpc-reverse-proxy-serving-cert has 13 months
```

Comment:
Any particular reason for this long rotation? What is the problem if we rotate these like all the remaining serving certs every 60/30 days? /cc @deads2k I'd like your input on this matter

Reply:
This applies to other certs as well.

Reply:
The only reason the localhost we have in kas-o are 10y validity is they are meant for recovery, but this one should be fine rotating like all the others with 60 validity and 30 refresh, imo.

Reply:
Since the front serving cert (kube-controller-manager svc)  of the proxy rotates after one year (service-ca operator), I thought  it might not be necessary to rotate the localhost certs faster. But I am not against changing it to any other value if we want to follow conventions.